Lib/test/test_subprocess.py

```diff
@@ -1661,6 +1662,78 @@ def test_start_new_session(self):
             child_pgid = int(output)
             self.assertNotEqual(parent_pgid, child_pgid)
 
+    def test_user(self):
+        # For code coverage of the user parameter.  We don't care if we get an
+        # EPERM error from it depending on the test execution environment, that
+        # still indicates that it was called.
+
+        uid = os.getuid()
+        for user in [65534 if uid != 65534 else 65533,
+                     "nobody" if sys.platform != 'darwin' else "unknown"]:
+            try:
+                output = subprocess.check_output(
+                        [sys.executable, "-c",
+                        "import os; print(os.getuid())"],
+                        user=user)
+            except OSError as e:
+                if e.errno != errno.EPERM:
+                    raise
+            else:
+                parent_user = os.getuid()
+                child_user = int(output)
+                self.assertNotEqual(parent_user, child_user)
+
+        with self.assertRaises(ValueError) as ex:
+            subprocess.check_call([sys.executable, "-c", "pass"], user=-1)
+
+    def test_groups(self):
+        gid = os.getgid()
+        group_list = [65534 if gid != 65534 else 65533,
+                      "nogroup" if sys.platform != 'darwin' else "staff"]
+
+        # first test numeric and symbolic primary group
+        for group in group_list:
+            try:
+                output = subprocess.check_output(
+                        [sys.executable, "-c",
+                        "import os; print(os.getgid())"],
+                        groups=[group])
+            except OSError as e:
+                if e.errno != errno.EPERM:
+                    raise
+            else:
+                parent_group = os.getgid()
+                child_group = int(output)
+                self.assertNotEqual(parent_group, child_group)
+
+        # make sure we bomb on negative values
+        with self.assertRaises(ValueError) as e:
+            subprocess.check_call([sys.executable, "-c", "pass"], groups=[-1])
+
+        # now primary and supplimental, use json so we can get a stable listing
+        try:
+            output = subprocess.check_output(
+                    [sys.executable, "-c",
+                    "import os, json; json.dumps(print(os.getgroupss()))"],
```

Comment:
You probably wanted `json.dumps(os.getgroups())`. As of now, this code always produces `null`, and this additionally shows that you should probably compare the output with the expected value instead of checking inequality with an unexpected one :)